packages/app/src/ui/pages/PresentVerifiableCredential/PresentVerifiableCredential.tsx

```diff
@@ -0,0 +1,169 @@
+
+import { usePresentVerifiableCredential } from "./usePresentVerifiableCredential";
+
+const PresentVerifiableCredential = (): JSX.Element => {
+  const {
+    isWalletConnected,
+    isWalletInstalled,
+    verifiablePresentationRequest,
+    cryptkeeperVerifiableCredentials,
+    selectedVerifiableCredentialHashes,
+    error,
+    isMenuOpen,
+    menuSelectedIndex,
+    menuRef,
+    onCloseModal,
+    onRejectRequest,
+    onToggleSelection,
+    onToggleMenu,
+    onMenuItemClick,
+    onSubmitVerifiablePresentation,
+  } = usePresentVerifiableCredential();
+
+  const menuOptions = [
+    isWalletConnected ? "Sign with Metamask" : "Connect to Metamask",
+    "Sign with Cryptkeeper",
+    "Proceed without Signing",
+  ];
+
+  return (
+    <Box
+      data-testid="select-verifiable-credential-page"
+      sx={{ display: "flex", flexDirection: "column", height: "100%" }}
+    >
+      <FullModalHeader onClose={onCloseModal}>Request for Verifiable Credentials</FullModalHeader>
+
+      <FullModalContent>
+        <Typography sx={{ textAlign: "center" }}>
+          You have received a request to present Verifiable Credentials:
+        </Typography>
+
+        <Typography sx={{ textAlign: "center" }}>{verifiablePresentationRequest}</Typography>
+
+        {cryptkeeperVerifiableCredentials.map(({ verifiableCredential, metadata }) => (
+          <VerifiableCredentialItem
+            key={metadata.hash}
+            metadata={metadata}
+            selected={selectedVerifiableCredentialHashes.includes(metadata.hash)}
+            verifiableCredential={verifiableCredential}
+            onToggleSelectVerifiableCredential={onToggleSelection}
+          />
+        ))}
+      </FullModalContent>
+
+      {error && (
+        <Typography color="error.main" fontSize="xs" sx={{ pb: 1 }} textAlign="center">
+          {error}
+        </Typography>
+      )}
+
+      <Box sx={{ p: "1rem", mt: "1rem", display: "flex", alignItems: "center" }}>
+        <Typography sx={{ mr: 1, textAlign: "center" }}>Choose Verifiable Credential signer</Typography>
+
+        <Tooltip
+          followCursor
+          title="Choose method of signing Verifiable Credentials. You may use your Cryptkeeper keys, connected Metamask, or send them unsigned."
+        >
+          <InfoIcon />
+        </Tooltip>
+      </Box>
+
+      <FullModalFooter>
+        <Box
+          sx={{ alignItems: "center", display: "flex", justifyContent: "space-between", width: "100%", height: "100%" }}
+        >
+          <Button
+            data-testid="reject-verifiable-presentation-request"
+            name="reject"
+            size="small"
+            sx={{ textTransform: "none", flex: 1, mr: 1, height: "2rem", width: "30%" }}
+            type="submit"
+            variant="outlined"
+            onClick={onRejectRequest}
+          >
+            Reject
+          </Button>
+
+          <ButtonGroup ref={menuRef} sx={{ height: "2rem", width: "70%" }} variant="contained">
```

Comment:
Thanks for adding the example how to use it with dropdown. I'll move this one in a separate component and reuse it for create identity.

Reply:
great!